torch_xla/csrc/reduction.cpp

```diff
@@ -422,6 +422,10 @@ xla::XlaOp BuildAll(xla::XlaOp input, absl::Span<const xla::int64> dimensions,
   xla::XlaOp result =
       xla::Reduce(input, init_value, CreateAllComputation(shape.element_type()),
                   dimensions);
+  xla::XlaOp zero = xla::Zero(input.builder(), shape.element_type());
```

Comment:
There is a minor optimization that might be worth doing, in order to minimize the HLO graph.
If the issue is happening only when the result has one element, checking the shape's number of elements will help in not adding superfluous HLO nodes.

Reply:
If all of the reduce dimension all have size 1, this issue will happen.

```
(Pdb) t1 = torch.tensor([[-147.5], [153.5]]).to(device)
(Pdb) print(torch.any(t1, dim=1))
tensor([-147.5000,  153.5000], device='xla:0')

(Pdb) t1 = torch.tensor([[-147.5]]).to(device)
(Pdb) print(torch.any(t1))
tensor(-147.5000, device='xla:0')
```

I added a check check to verify that.